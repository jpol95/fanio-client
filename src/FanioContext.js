import React from 'react'

const FanioContext = React.createContext({
    fandomList: [],
    reviewList: [],
    sectionList: [], 
    subList: [],
    tagList: [],
    reviewTagList: [],
    installmentList: [],
    handleSubmitReview: () => {}, 
    handleAddFandom: () => {}, 
    handleSubmitInstallments: () => {}, 
    handleGetReview: () => {}
})

export default FanioContext