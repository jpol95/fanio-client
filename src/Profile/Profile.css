@import url('https://fonts.googleapis.com/css2?family=Audiowide&display=swap');
.divider{
    width: 100%;
}
.label {
    width: 100%;
}

.profile-label{
    font-family: 'Audiowide', cursive;
    font-size: 50px
}

@media only screen and (min-width: 800px){
    .profile-label{
        font-family: 'Audiowide', cursive;
        font-size: 100px
    }
}