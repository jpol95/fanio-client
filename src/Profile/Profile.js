import React from "react";
import PersonalInfo from "../PersonalInfo/PersonalInfo";
import Reviews from "../Reviews/Reviews";
import Fandom from "../Fandom/Fandom";
import FanioContext from "../FanioContext";
import "./Profile.css";
import { Link } from "react-router-dom";
import FetchService from '../FetchService'

export default class Profile extends React.Component {
  static contextType = FanioContext;

  componentDidMount = () => {
    this.props.fetchFandoms()
  }


  getFandoms() {
    // console.log(this.context.fandomList)
    return this.context.fandomList.map(fandom => {
      return <Fandom key={fandom.id} {...fandom} />
    })
  }


  render() {

    return (
      <React.Fragment>
        <PersonalInfo />
        <Reviews />
        {this.getFandoms()}
        <Link to={`/users/1/fandom-form`}>Create Fandom</Link>
      </React.Fragment>
    );
  }
}
