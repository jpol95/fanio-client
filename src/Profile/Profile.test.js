import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Profile from './Profile';
import FanioContext from '../FanioContext'
import dummyStore from '../dummy-store'
import {BrowserRouter} from 'react-router-dom'

describe('Create Installment component', () => {
    const props = {
    match: {
        params: {
            userId: 1, 
            fandomId: 5, 
            installmentId: 6, 
            sectionId: 1, 
        }
    }, 
    installId: 1
}


  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><FanioContext.Provider value={{reviewList: dummyStore.reviewList, fandomList: dummyStore.fandomList, installmentList:dummyStore.installmentList, loggedInUser: 1, sectionList: dummyStore.sectionList}}> <Profile {...props} /> </FanioContext.Provider></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><FanioContext.Provider value={{reviewList: dummyStore.reviewList, fandomList: dummyStore.fandomList, installmentList:dummyStore.installmentList, loggedInUser: 1, sectionList: dummyStore.sectionList}}> <Profile {...props} /> </FanioContext.Provider></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});