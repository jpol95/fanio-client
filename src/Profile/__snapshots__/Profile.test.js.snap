// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders the UI as expected 1`] = `
Array [
  " ",
  <div
    className="profile-label"
  >
    PROFILE
  </div>,
  <div
    className="profile-label"
  >
    FANDOMS
  </div>,
  <img
    alt="superhero divider"
    className="divider"
    src="divider2.PNG"
  />,
  <a
    className="fandom-view"
    href="/users/1/fandom-view/1"
    onClick={[Function]}
  >
    <h3>
      Doctor Who
    </h3>
    <button
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </a>,
  <a
    className="fandom-view"
    href="/users/1/fandom-view/2"
    onClick={[Function]}
  >
    <h3>
      Supernatural
    </h3>
    <button
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </a>,
  <a
    className="fandom-view"
    href="/users/1/fandom-view/3"
    onClick={[Function]}
  >
    <h3>
      Parks and Recreation
    </h3>
    <button
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </a>,
  <a
    className="fandom-view"
    href="/users/1/fandom-view/4"
    onClick={[Function]}
  >
    <h3>
      Buffy the Vampire Slayer
    </h3>
    <button
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </a>,
  <a
    className="fandom-view"
    href="/users/1/fandom-view/5"
    onClick={[Function]}
  >
    <h3>
      Harry Potter
    </h3>
    <button
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </a>,
  <a
    className="fandom-view"
    href="/users/1/fandom-view/6"
    onClick={[Function]}
  >
    <h3>
      This belongs to user 2
    </h3>
    <button
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </a>,
  <a
    className="create-fandom-button"
    href="/users/1/create-fandom"
    onClick={[Function]}
  >
    Create Fandom
  </a>,
  " ",
]
`;
